src/Sulu/Bundle/ContactBundle/Tests/Functional/Infrastructure/Sulu/Search/AccountReindexProviderTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContactBundle\Tests\Functional\Infrastructure\Sulu\Search;
+
+use CmsIg\Seal\Reindex\ReindexConfig;
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\ContactBundle\Entity\Account;
+use Sulu\Bundle\ContactBundle\Entity\AccountInterface;
+use Sulu\Bundle\ContactBundle\Infrastructure\Sulu\Search\AccountReindexProvider;
+use Sulu\Bundle\TestBundle\Testing\SuluTestCase;
+
+class AccountReindexProviderTest extends SuluTestCase
+{
+    private EntityManagerInterface $entityManager;
+    private AccountReindexProvider $provider;
+
+    protected function setUp(): void
+    {
+        $this->entityManager = $this->getEntityManager();
+        $this->provider = new AccountReindexProvider($this->entityManager);
+        $this->purgeDatabase();
+    }
+
+    public function testGetIndex(): void
+    {
+        $this->assertSame('admin', AccountReindexProvider::getIndex());
+    }
+
+    public function testTotal(): void
+    {
+        $this->assertNull($this->provider->total());
+    }
+
+    public function testProvideWithoutIdentifiers(): void
+    {
+        $account1 = $this->createAccount('Test Account 1');
+        $account2 = $this->createAccount('Test Account 2');
+
+        $this->entityManager->flush();
+
+        $config = ReindexConfig::create()->withIndex('admin');
+        $results = \iterator_to_array($this->provider->provide($config));
+
+        $this->assertCount(2, $results);
+
+        $this->assertSame(AccountInterface::RESOURCE_KEY . '::' . $account1->getId(), $results[0]['id']);
+        $this->assertSame(AccountInterface::RESOURCE_KEY, $results[0]['resourceKey']);
+        $this->assertSame((string) $account1->getId(), $results[0]['resourceId']);
+        $this->assertSame('Test Account 1', $results[0]['title']);
+        $this->assertNotNull($results[0]['createdAt']);
+
+        $this->assertSame(AccountInterface::RESOURCE_KEY . '::' . $account2->getId(), $results[1]['id']);
+        $this->assertSame('Test Account 2', $results[1]['title']);
```

Comment:
I would simplify the test:

Do not match every single property match against the whole results:

```suggestion
        $this->assertSame(
            [
                [
                     'id' => ...
                ],
            ],
            [...$result]
        );
```